programs/benchmark/Benchmark.cpp

```diff
@@ -342,14 +345,33 @@ class Benchmark : public Poco::Util::Application
             }
         }
 
-        pool.wait();
-        total_watch.stop();
+        // Setup signal handler to capture additional SIGINTs (in case the user wants to quit immediately).
+        signal(SIGINT, signalHandler);
+
+        /// run the interrupt handler thread in background detached and
+        /// block if and until signalHandler notifies and then exit.
+        auto interrupt_handler = std::thread{[&]() {
+            while (!cancel_benchmark)
+            {
+            }
+            std::cout << "Second SIGINT received. Quitting immediately."
+                      << "\n";
+            _exit(1);
+        }};
+        interrupt_handler.detach();
```

Comment:
Oops... that's extremely dangerous.

There is almost no good usage for detached threads in C++.
And there is a race condition if the program start to exit and destroy global variables and you call std::cerr.